File: program/find-the-maximum-sum-of-a-contiguous-subarray/FindTheMaximumSumOfAContiguousSubarray.java
public class FindTheMaximumSumOfAContiguousSubarray {
    public static void main(String[] args) {
        int[] numberArray = {-2, -3, 4, -1, -2, 1, 5, -3};
        System.out.println("Maximum contiguous sum is "
          + maxSubArraySum(numberArray));
      }
    
      static int maxSubArraySum(int numberArray[]) {
        int size = numberArray.length;
        int maximum_so_far = Integer.MIN_VALUE, maximum_ending_here = 0;
    
        for (int i = 0; i < size; i++) {
          maximum_ending_here = maximum_ending_here + numberArray[i];
          if (maximum_so_far < maximum_ending_here)
            maximum_so_far = maximum_ending_here;
          if (maximum_ending_here < 0)
            maximum_ending_here = 0;
        }
        return maximum_so_far;
      }
    
}

File: program/program/find-the-maximum-sum-of-a-contiguous-subarray-(kadane's-algorithm)/find_the_maximum_sum_of_a_contiguous_subarray.php
<?php

function maxSubArraySum($a, $size)
{
	$max_so_far = PHP_INT_MIN;
	$max_ending_here = 0;

	for ($i = 0; $i < $size; $i++)
	{
		$max_ending_here = $max_ending_here + $a[$i];
		if ($max_so_far < $max_ending_here)
			$max_so_far = $max_ending_here;

		if ($max_ending_here < 0)
			$max_ending_here = 0;
	}
	return $max_so_far;
}

$a = array(-2, -3, 4, -1, -2, 1, 5, -3);
$n = count($a);
$max_sum = maxSubArraySum($a, $n);
echo "Maximum sum of contiguous subarray : " ,$max_sum;

?>
